app/models/manageiq/providers/ansible_tower/shared/automation_manager/tower_api.rb

```diff
@@ -21,6 +23,9 @@ def create_in_provider(manager_id, params)
     def create_in_provider_queue(manager_id, params, auth_user = nil)
       process_secrets(params) if respond_to?(:process_secrets)
       manager = ExtManagementSystem.find(manager_id)
+      if ManageIQ::Providers::Inflector.provider_name(manager) != ManageIQ::Providers::Inflector.provider_name(self)
```

Comment:
It feels strange that we've gotten this far without an error.  I know that we want the create to happen in the provider first and then refresh to bring the record into our database, but should this live as a validation on the model instead?  @Fryguy Any thoughts on where this belongs?

Reply:
This only happens through calls to API since calls through API allow user to provide arbitrary `manager_id`

Reply:
Right, but adding an Ansible credential type to a Foreman provider doesn't make sense on either system, that's why I was suggesting the validation on the model.

Reply:
The problem here is that it is doing an ExtManagementSystem.find instead of the more targeted find of only valid EMSes.